st3/lsp_utils/npm_client_handler.py

```diff
@@ -0,0 +1,146 @@
+import shutil
+import sublime
+
+from LSP.plugin.core.handlers import LanguageHandler
+from LSP.plugin.core.protocol import Response
+from LSP.plugin.core.settings import ClientConfig, read_client_config
+from LSP.plugin.core.typing import Callable, Dict
+from .server_npm_resource import ServerNpmResource
+
+CLIENT_SETTING_KEYS = ['env', 'experimental_capabilities', 'languages', 'initializationOptions', 'settings']
+
+
+class ApiWrapper(object):
+    def __init__(self, client):
+        self.__client = client
+
+    def on_notification(self, method: str, handler: Callable) -> None:
+        self.__client.on_notification(method, handler)
+
+    def on_request(self, method: str, handler: Callable) -> None:
+        self.__client.on_request(
+            method,
+            lambda params, request_id: self.__client.send_response(Response(request_id, handler(params))))
+
+
+class NpmClientHandler(LanguageHandler):
+    # To be overridden by subclass.
+    PACKAGE_NAME = None
+    SERVER_DIRECTORY = None
+    SERVER_BINARY_PATH = None
+    # Internal
+    __SERVER = None
```

Comment:
Why capital letters? It is an object/instance, isn't it?

Reply:
yes, was just thinking about it